### DISCUSSION TITLE
Replicating a database API

### DISCUSSION THREAD

**Discussion by usagibear** [mockoon/mockoon#819] [Help] [answered]
Hello,
I am new to testing and am testing a .NET 6 web API project with Nunit. I am using Mockoon to mock the API calls for these tests. I want to be able to replicate an in memory database like with DbContext so that when I call a Mockoon Endpoint to create a user (`POST` to `/api/user`) that when I get a list of users (`GET` to `/api/user`), I will get a list of all the previous users and this newly added one. When the environment is restarted, everything should be reset. 

Currently, when I use the templating system, I can get all users but when I create a new user, this user does not exist when I run the get all users call again.

Is this possible in Mockoon at the moment? If so, how can this be done? Is this a good practice?

I wanted to do this type of testing because I need to test if certain functions have the correct effect.

I am primarily following this [article](https://mockoon.com/tutorials/generate-mock-json-data/).

If you have any questions, concerns, or comments, please let me know. Thanks!

**Comment by 255kb** [comment, marked as answer]
Hi, 
Currently, Mockoon's routes are stateless and independent from each others. What you are looking for are [CRUD routes](https://github.com/mockoon/mockoon/issues/124) that are not available yet. 
In the meantime, we are working on a system of [persisting data buckets](https://github.com/mockoon/mockoon/issues/783) for the next release. This is one of the prerequisites for CRUD routes. Data buckets will be generated when the server starts and be reset when it is stopped. 

I hope its clearer now. Stay tuned!

**Reply by usagibear** [reply]
Thank you for the reply and the hard work. I'll keep my eyes out for it.
